src/dialects/ibmi/index.js

```diff
@@ -39,6 +39,10 @@ export class IBMiDialect extends AbstractDialect {
   constructor(sequelize) {
     super();
     this.sequelize = sequelize;
+
+    // Attach DSN configuation to the Sequelize object
+    this.sequelize.config.DSN = sequelize.options.DSN;
```

Comment:
Instanciating the dialect shouldn't cause any side effect. If DSN is an option specific to ibmi, I think it should go in `dialectOptions` instead.